base-lite/ub/ub.go

```diff
@@ -0,0 +1,313 @@
+		if len(parts) == 2 {
+			m[parts[0]] = parts[1]
+		}
+	}
+	return m
+}
+
+func splitToMapDefaults(separator string, defaultValues string, value string) map[string]string {
+	values := KvStringToMap(defaultValues, separator)
+	for k, v := range KvStringToMap(value, separator) {
+		values[k] = v
+	}
+	return values
+}
+
+func KvStringToMap(kvString string, sep string) map[string]string {
+	return ListToMap(strings.Split(kvString, sep))
+}
+
+// GetEnvironment returns the current environment as a map.
+func GetEnvironment() map[string]string {
+	return ListToMap(os.Environ())
+}
+
+// buildProperties creates a map suitable to be output as Java properties from a ConfigSpec and a map representing an environment.
+func buildProperties(spec ConfigSpec, environment map[string]string) map[string]string {
+	config := make(map[string]string)
+	for key, value := range spec.Defaults {
+		config[key] = value
+	}
+
+	for envKey, envValue := range environment {
+		if newKey, found := spec.Renamed[envKey]; found {
+			config[newKey] = envValue
+		} else {
+			if !Contains(spec.Excludes, envKey) && !(len(spec.ExcludeWithPrefix) > 0 && strings.HasPrefix(envKey, spec.ExcludeWithPrefix)) {
+				for prefix, keep := range spec.Prefixes {
+					if strings.HasPrefix(envKey, prefix) {
+						var effectiveKey string
+						if keep {
+							effectiveKey = envKey
+						} else {
+							effectiveKey = envKey[len(prefix)+1:]
+						}
+						config[ConvertKey(effectiveKey)] = envValue
+					}
+				}
+			}
+		}
+	}
+	return config
+}
+
+func writeConfig(writer io.Writer, config map[string]string) bool {
+	// Go randomizes iterations over map by design. We sort properties by name to ease debugging:
+	sortedNames := make([]string, 0, len(config))
+	for name := range config {
+		sortedNames = append(sortedNames, name)
+	}
+	sort.Strings(sortedNames)
+	for _, n := range sortedNames {
+		_, err := fmt.Fprintf(writer, "%s=%s\n", n, config[n])
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error printing configs: %s", err)
+			return false
+		}
+	}
+	return true
+}
+
+func loadConfigSpec(path string) (ConfigSpec, bool) {
+	var spec ConfigSpec
+	jsonFile, err := os.Open(path)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error opening json file %s : %s", path, err)
+		return spec, false
+	}
+	bytes, err := io.ReadAll(jsonFile)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error reading from json file %s : %s", path, err)
+		return spec, false
+	}
+
+	errParse := json.Unmarshal(bytes, &spec)
+	if errParse != nil {
+		fmt.Fprintf(os.Stderr, "Error parsing json file %s : %s", path, errParse)
+	}
+	return spec, true
+}
+
+func invokeJavaCommand(className string, jvmOpts string, args []string) bool {
+	classPath := getEnvOrDefault("UB_CLASSPATH", "/usr/share/java/cp-base-lite/*")
+
+	opts := []string{}
+	if jvmOpts != "" {
+		opts = append(opts, jvmOpts)
+	}
+	opts = append(opts, "-cp", classPath, className)
+	cmd := exec.Command("java", append(opts[:], args...)...)
```

Comment:
Do you want to hide the stdout/stderr? That's what this will currently do, you may want to:
```go
cmd.Stdout = os.Stdout
cmd.Stderr = os.Stderr
```